carbonmark/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,149 @@
+    thumbsApi.scrollPrev();
+  }, [mainApi, thumbsApi]);
+
+  const onNextClick = useCallback(() => {
+    if (!mainApi || !thumbsApi) return;
+    mainApi.scrollNext();
+    thumbsApi.scrollNext();
+  }, [mainApi, thumbsApi]);
+
+  const onSelect = useCallback(() => {
+    if (!mainApi || !thumbsApi) return;
+    setSelectedIndex(mainApi.selectedScrollSnap());
+    thumbsApi.scrollTo(mainApi.selectedScrollSnap());
+  }, [mainApi, thumbsApi, setSelectedIndex]);
+
+  useEffect(() => {
+    if (!mainApi) return;
+    onSelect();
+    mainApi.on("select", onSelect);
+    mainApi.on("reInit", onSelect);
+  }, [mainApi, onSelect]);
+
+  useEffect(() => {
+    if (props.location) {
+      const images = [...props.images];
+      images.unshift({
+        url: "/map-thumbnail.png",
+        caption: images?.[1]?.caption,
+      });
+      setImages(images);
+    }
+  }, [props.location, props.images]);
+
+  return (
+    <div className={styles.carousel}>
+      <div ref={mainRef} className={styles.viewport}>
+        <div className={styles.container}>
+          {images?.map((image: CarouselImage, index: number) => (
+            <div className={styles.slide} key={`slide-${index}`}>
+              {props.location && index === 0 ? (
+                <ProjectMap
+                  zoom={5}
+                  lat={props.location?.geometry.coordinates[1]}
+                  lng={props.location?.geometry.coordinates[0]}
+                />
+              ) : (
+                <>
+                  <TextInfoTooltip tooltip={image.caption}>
+                    <div className="icon-wrapper">
+                      <InfoOutlinedIcon width={20} height={20} />
+                    </div>
+                  </TextInfoTooltip>
+                  <img
+                    src={image.url}
+                    alt={image.caption}
+                    className={styles.slideImg}
+                  />
+                </>
+              )}
+            </div>
+          ))}
+        </div>
+      </div>
+      <div className={styles.thumbs}>
+        {!isMobile &&
+          images?.length > maxSlidesToShow &&
+          selectedIndex !== 0 && (
+            <div
+              onClick={onPreviousClick}
+              className={cx(styles.arrows, "left")}
+            >
+              <KeyboardArrowLeftOutlinedIcon />
+            </div>
+          )}
+        <div ref={thumbsRef} className={styles.thumbsViewport}>
+          <div className={styles.thumbsContainer}>
+            {images?.map((image: CarouselImage, index: number) => (
+              <div key={`thumbnail-${index}`} className={styles.thumbsSlide}>
+                <button
+                  type="button"
+                  onClick={() => onThumbClick(index)}
+                  className={styles.thumbsSlideButton}
+                >
+                  {index === selectedIndex && (
+                    <div className={styles.selected} />
+                  )}
+                  <img
```

Comment:
I believe Next `Image` would work here, which could save quite a bit of bandwidth if we are able to use the `sizes` prop to constrain the max image size intelligently.
We have a util to make that more straightforward, lmk if you need help:
https://github.com/KlimaDAO/klimadao/blob/96db714b4ada84a4e3e3a31fe8301c2db3c68be7/lib/utils/getImageSizes/index.ts#L14

This `sanityImageLoader` should work
https://github.com/KlimaDAO/klimadao/blob/cac833f412c8a9f183f4dbc49e5faf551ce7e920/site/lib/sanityImageLoader.ts#L3

https://nextjs.org/docs/pages/api-reference/components/image#loader